src/main/java/seedu/contax/model/person/Person.java

```diff
@@ -60,6 +60,10 @@ public Address getAddress() {
         return Collections.unmodifiableSet(tags);
     }
 
+    public Person updateTags(Set<Tag> newTags) {
+        return new Person(name, phone, email, address, newTags);
+    }
```

Comment:
To save you the trouble (because the getter returns an immutable set), you could  change this to `withTag(Tag t)` and `withoutTag(Tag t)` functions to make the logic code cleaner.

Reply:
Thank you for the suggestion!

Updated and fixed in 74d817e